mapie/estimator/estimator.py

```diff
@@ -434,8 +436,9 @@ def fit(
                     )
                     for train_index, _ in cv.split(X)
                 )
-            if isinstance(cv, ShuffleSplit):
-                single_estimator_ = estimators_[0]
+                # In split-CP, we keep only the model fitted on train dataset
+                if check_no_agg_cv(cv, self.no_agg_cv_):
```

Comment:
check_no_agg_cv is used multiple times and always return the same results. I suggest that you define its value once and for all : 
```python
self.cv_has_no_agg = check_no_agg_cv(cv, self.no_agg_cv)
```

Reply:
This is also the case when we test `self.method in self.no_agg_methods_` for example. For the sake of clarity, I prefer to leave it without using the object attribute and avoid multiplying object attributes that are unnecessary for the user in particular.

Reply:
fine